### PR TITLE
Declare encoding/decoding between Bytes and String in domains.md

### LINKED ISSUE #1953: Add support for unicode encoding/decoding between Bytes and String
It would be nice to be able to encode/decode UTF-8 strings for being able to print (in the proof configuration) error messages associated wiht Solidity's ABI encoding https://docs.soliditylang.org/en/v0.5.3/abi-spec.html#types.

Proposed signature:

```
syntax String ::= decodeBytes ( encoding: String , contents: Bytes ) [function, hook(BYTES.decodeBytes)]
syntax Bytes ::= encodeBytes ( encoding: String , contents: String ) [function, hook(BYTES.encodeBytes)]
```

### PR DESCRIPTION
**Pull request by andreiburdusa**
Fixes #1953

### CODE REVIEW

File: k-distribution/include/kframework/builtin/domains.md [hunk 1]
```diff
@@ -1661,10 +1661,31 @@ divergent behavior unless the `Bytes` type is used in a manner that preserves
...
+endmodule
+```
+
+```k
+module BYTES-STRING-ENCODE [kore, symbolic]
+  imports BYTES-SYNTAX
+```
+### Encoding/decoding between Bytes and String
+
+You can encode/decode between Bytes and String using`UTF-8`, `UTF-16LE`, `UTF-16BE`, `UTF-32LE`, and `UTF-32BE`
```
Comment (ttuegel) [inline k-distribution/include/kframework/builtin/domains.md]:
```suggestion
You can encode/decode between Bytes and String using `UTF-8`, `UTF-16LE`, `UTF-16BE`, `UTF-32LE`, and `UTF-32BE`
```

File: k-distribution/include/kframework/builtin/domains.md [hunk 2]
```diff
@@ -1661,10 +1661,31 @@ divergent behavior unless the `Bytes` type is used in a manner that preserves
...
+```k
+    syntax String ::= decodeBytes ( encoding: String , contents: Bytes ) [function, hook(BYTES.decodeBytes)]
+    syntax Bytes ::= encodeBytes ( encoding: String , contents: String ) [function, hook(BYTES.encodeBytes)]
+endmodule
+```
+
+```k
+module BYTES-HOOKED
+  imports STRING-SYNTAX
+  imports BYTES-SYNTAX
```
Comment (ttuegel) [inline k-distribution/include/kframework/builtin/domains.md:1688]: The new module needs to be included by default: 

```suggestion
  imports BYTES-SYNTAX
  imports BYTES-STRING-ENCODE
```

**Review by ttuegel — CHANGES_REQUESTED** [review]

**Review by dwightguth — COMMENTED** [review]
Wouldn't it be better if we supported any encoding that was supported by `libiconv`?

**Comment by ttuegel** [comment]
> Wouldn't it be better if we supported any encoding that was supported by `libiconv`?

The only binding I can find for `libiconv` does not seem to be maintained. The original request was only for UTF-8, so I'm inclined to approve it as-is until we need more.

**Review by ttuegel — APPROVED** [review]

**Review by dwightguth — APPROVED** [review]

**Pull request merged**